tests/test_application.py

```diff
@@ -39,6 +39,30 @@ def test_extract_multiple_tag_enclosed_certs(self):
         self.assertEqual(["my_cert1", "my_cert2"], extract_certs(pem))
 
 
+class TestEncryptedKeyAsClientCredential(unittest.TestCase):
+    # Internally, we use serialization.load_pem_private_key() to load an encrypted private key with a passphrase
+    # This function takes in encrypted key in bytes and passphrase in bytes too
+    # Our code handles such a conversion, adding test cases to verify such a conversion is needed
+
+    def test_encyrpted_key_in_bytes_and_string_password_should_error(self):
+        private_key = b"""
+        -----BEGIN ENCRYPTED PRIVATE KEY-----
+        test_private_key
+        -----END ENCRYPTED PRIVATE KEY-----
+        """
+        self.assertRaises(TypeError, serialization.load_pem_private_key,
+                          private_key, password="string_password", backend=default_backend())
```

Comment:
Nit: would suggest to use a more modern style to write this.
```suggestion
        self.assertRaises(TypeError):
            serialization.load_pem_private_key(
                private_key, password="string_password", backend=default_backend())
```